consumer/maxer.go

```diff
@@ -16,11 +18,17 @@ func NewMaxer(stopOnFirst bool) Consumer {
 
 // Run against an input channel until it is closed..
 func (m *Maxer) Run(in <-chan int, done chan<- bool) {
+	defer func() {
+		done <- true
```

Comment:
I *think* this can be a `close(done)`. Additionally, maybe this would be best placed in a (private) helper function:

```go
func channelCleanup(in <-chan int, done chan<- bool) {
    close(done)
    for range in {
        // do nothing
    }
}
```